functionary/ui/forms/scheduled_task.py

```diff
@@ -36,59 +36,70 @@ class ScheduledTaskForm(ModelForm):
         initial="*",
         validators=[month_of_year_validator],
     )
+    function = ModelChoiceField(queryset=Function.objects.all(), required=True)
 
     class Meta:
         model = ScheduledTask
+
+        # NOTE: The order of the fields matters. The clean_<field> methods run based on
+        # the order they are defined in the 'fields =' attribute
         fields = [
             "name",
+            "environment",
             "description",
             "status",
             "function",
             "parameters",
-            "environment",
         ]
 
     def __init__(self, *args, **kwargs):
-        env: Environment = kwargs.pop("env")
+        environment = kwargs.pop("environment", None)
```

Comment:
Somethings I probably should have noticed before, so apologies:

It'd probably be better to just list environment in the signature of the `__init__` and then not pass it along when you call super:

```python
def __init__(self, environment: Environment, *args, **kwargs):
    # Now this call doesn't include environment because it's not bundled in with **kargs
    super().__init__(*args, **kwargs)
```

As for `is_create`, I don't think that should be necessary as a parameter.  That info is implicit in the presence or absence of the `instance` parameter.  When doing an update, that would be set to the instance you are updating.  If that's not set, then you are creating a new instance.

Reply:
Fixed